labelme/labelDialog.py

```diff
@@ -36,22 +36,38 @@
 
 class LabelDialog(QDialog):
 
-    def __init__(self, text="Enter object label", parent=None):
+    def __init__(self, text="Enter object label", parent=None, labels=None):
         super(LabelDialog, self).__init__(parent)
         self.edit = QLineEdit()
-        self.edit.setText(text)
+        self.edit.setPlaceholderText(text)
         self.edit.setValidator(labelValidator())
         self.edit.editingFinished.connect(self.postProcess)
         layout = QVBoxLayout()
         layout.addWidget(self.edit)
+        # buttons
         self.buttonBox = bb = BB(BB.Ok | BB.Cancel, Qt.Horizontal, self)
         bb.button(BB.Ok).setIcon(newIcon('done'))
         bb.button(BB.Cancel).setIcon(newIcon('undo'))
         bb.accepted.connect(self.validate)
         bb.rejected.connect(self.reject)
         layout.addWidget(bb)
+        # label_list
+        self.labelList = QListWidget()
+        if labels:
+            self.labelList.addItems(labels)
+        self.labelList.currentItemChanged.connect(self.labelSelected)
+        layout.addWidget(self.labelList)
         self.setLayout(layout)
 
+    def addLabelHistory(self, label):
+        if self.labelList.findItems(label, Qt.MatchExactly):
+            return
+        self.labelList.addItem(label)
+        self.labelList.sortItems()
```

Comment:
Is this a good idea? Shouldn't the labels rather be sortable by the user? (e.g. most common first)
And keep the order of the CLI?

Reply:
I think it is easier for the annotator to select if label list is sorted, so sorting is better in default.
And if requested, I can add an option like `--nosortlabels` to disable this sorting.
What do you think?